middleware/http/oauth2/oauth2_middleware.go

```diff
@@ -15,144 +15,299 @@ package oauth2
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"net/http"
 	"net/url"
 	"reflect"
-	"strings"
+	"time"
 
-	"github.com/fasthttp-contrib/sessions"
 	"github.com/google/uuid"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwe"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/spf13/cast"
 	"golang.org/x/oauth2"
 
 	"github.com/dapr/components-contrib/internal/httputils"
-	"github.com/dapr/components-contrib/internal/utils"
 	mdutils "github.com/dapr/components-contrib/metadata"
 	"github.com/dapr/components-contrib/middleware"
 	"github.com/dapr/kit/logger"
 )
 
-// Metadata is the oAuth middleware config.
-type oAuth2MiddlewareMetadata struct {
-	ClientID       string `json:"clientID" mapstructure:"clientID"`
-	ClientSecret   string `json:"clientSecret" mapstructure:"clientSecret"`
-	Scopes         string `json:"scopes" mapstructure:"scopes"`
-	AuthURL        string `json:"authURL" mapstructure:"authURL"`
-	TokenURL       string `json:"tokenURL" mapstructure:"tokenURL"`
-	AuthHeaderName string `json:"authHeaderName" mapstructure:"authHeaderName"`
-	RedirectURL    string `json:"redirectURL" mapstructure:"redirectURL"`
-	ForceHTTPS     string `json:"forceHTTPS" mapstructure:"forceHTTPS"`
-}
+const (
+	// Timeout for authenticating with the IdP
+	authenticationTimeout = 10 * time.Minute
```

Comment:
Yes it's a timeout on authentication requests. Because we cannot influence the timeout on the IdP, we can only set an expiration on the cookie with the state token.

I am not sure this needs to be configurable. 10 mins should be plenty for someone to authenticate.